storage/innobase/dict/dict0crea.cc

```diff
@@ -133,7 +133,8 @@ dict_create_sys_tables_tuple(
 	ptr = static_cast<byte*>(mem_heap_alloc(heap, 4));
 	/* Be sure all non-used bits are zero. */
 	ut_a(!(table->flags2 & DICT_TF2_UNUSED_BIT_MASK));
-	mach_write_to_4(ptr, table->flags2);
+	/* create a new table, table->n_cols_core should be always 0 in tuple */
+	mach_write_to_4(ptr, dict_table_encode_mix_len(table->flags2, 0));
```

Comment:
I think that it would be better to write like this:
```
/* When creating a new table, there is no history of instant ADD COLUMN. */
ut_ad(table->n_cols_core == 0);
table->encode_mix_len(ptr);
```